src/main/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAligner.scala

```diff
@@ -0,0 +1,187 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.alignment
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.alignment.NeedlemanWunschAligner._
+import htsjdk.samtools.CigarOperator
+
+import scala.collection.mutable
+import scala.math.max
+
+object NeedlemanWunschAligner {
+  /** Generates a simple scoring function using the match and mismatch scores. */
+  def simpleScoringFunction(matchScore: Int, mismatchScore: Int): (Byte, Byte) => Int = {
+    (lhs: Byte, rhs: Byte) => if (lhs == rhs) matchScore else mismatchScore
+  }
+
+  /** Creates a NW aligner with fixed match and mismatch scores. */
+  def apply(matchScore: Int, mismatchScore: Int, gapOpen: Int, gapExtend: Int): NeedlemanWunschAligner = {
+    new NeedlemanWunschAligner(scoringFunction=simpleScoringFunction(matchScore, mismatchScore), gapOpen, gapExtend)
+  }
+
+  /** Directions within the trace back matrix. */
+  type Direction = Byte
+  val Left: Direction     = 1.toByte
+  val Up  : Direction     = 2.toByte
+  val Diagonal: Direction = 3.toByte
```

Comment:
Or, you know, I could make it `Leftward`, `Upwards` and `Diagonal` to get them as close as possible to the same length :p